pkg/reconcilers/deploymentconfig.go

```diff
@@ -57,6 +57,77 @@ func GenericBackendMutators() []DCMutateFn {
 	}
 }
 
+func DeploymentConfigArgsMutator(_, existing *appsv1.DeploymentConfig) (bool, error) {
+	update := true
+	existing.Spec.Template.Spec.Containers[0].Args = []string{"bin/3scale_backend", "-s", "falcon", "start", "-e", "production", "-p", "3000", "-x", "/dev/stdout"}
+	return update, nil
```

Comment:
is this always updating the dc on cluster? can we somehow check if the args already are in the expected state?

Reply:
Agree this would be better

Reply:
```go
func DeploymentConfigListenerArgsMutator(_, existing *appsv1.DeploymentConfig) (bool, error) {
	update := true
	falconArgs := []string{"bin/3scale_backend", "-s", "falcon", "start", "-e", "production", "-p", "3000", "-x", "/dev/stdout"}
	if reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].Args, falconArgs) {
		existing.Spec.Template.Spec.Containers[0].Args = falconArgs
		return update, nil
	}
	update = false
	return update, nil
}
```

Reply:
Will see if this works

Reply:
nope it doesn't

Reply:
should have used  `!reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].Args, falconArgs)` :facepalm: